File: Golang/problems/easy/problem476/solution.go
package solutions

// FindComplement function is a solution for the
// Problem #476 - Number Complement
// from leetcode.com
func FindComplement(num int) int {

	bits := []byte{}
	for num > 0 {
		reminder := num % 2
		bits = append(bits, byte(reminder)^1)

		num = num / 2
	}

	value, power := 0, 1
	for i := 0; i < len(bits); i++ {
		value += int(bits[i]) * power
		power *= 2
	}

	return value
}

File: Java/src/problems/easy/problem476/Solution.java
package problems.easy.problem476;

import utils.Helper;

public class Solution {
    public int findComplement(int num) {
        var result = 0;
        var mask = 1;

        while (num != 0) {
            result += ((num & 1) ^ 1) * mask;
            num >>>= 1;
            mask <<= 1;
        }
        return result;
    }
}
